guildwars2/guild/general.py

```diff
@@ -213,9 +247,66 @@ class GeneralGuild:
                         inline=False)
                     counter += 1
         if counter == 0:
-            return await ctx.send(
-                "No stash log entries yet for {}".format(guild_name.title()))
+            return await ctx.send("No stash log entries yet for {}".format(
+                guild_name.title()))
         try:
             await ctx.send(embed=data)
         except discord.Forbidden:
             await ctx.send("Need permission to embed links")
+
+    @guild.command(name="default", usage="<guild name>")
+    @commands.guild_only()
+    @commands.cooldown(1, 10, BucketType.user)
+    async def guild_default(self, ctx, *, guild_name=None):
+        """ Set your preferred guild for guild commands"""
+
+        guild = ctx.guild
+        if guild_name is None:
+            await self.bot.database.set_guild(guild, {
+                "guild_ingame": None,
+            }, self)
+            await ctx.send("Your preferred guild is now reset for this server."
+                           )
+        else:
+            guild_id = await self.guildname_to_guildid(ctx, guild_name)
+            # Write to DB, overwrites existing guild
+            await self.bot.database.set_guild(guild, {
+                "guild_ingame": guild_id,
+            }, self)
+
+            await ctx.send(
+                "Your preferred guild is now set to {0} for this server"
+                .format(guild_name))
+
+    async def get_preferred_guild(self, ctx):
+        # guild in this case is the server ID
+        guild = ctx.guild
+        if guild:
+            doc = await self.bot.database.get_guild(guild, self) or {}
+            return doc.get("guild_ingame")
+        else:
+            return None
+
+    async def guildid_to_guildname(self, ctx, guild_id):
+        try:
+            endpoint_name = "guild/{0}".format(guild_id)
+            results = await self.call_api(endpoint_name)
+            guild_name = results["name"]
+            return guild_name
+        except APIError as e:
+            return await self.error_handler(ctx, e)
+
+    async def guildname_to_guildid(self, ctx, guild_name):
+        try:
+            endpoint_id = "guild/search?name=" + guild_name.replace(' ', '%20')
+            guild_id = await self.call_api(endpoint_id)
+            guild_id = guild_id[0]
+            return guild_id
+        except (IndexError, APINotFound):
+            return await ctx.send("Invalid guild name")
+        except APIForbidden:
+            return await ctx.send(
+                "You don't have enough permissions in game to "
+                "use this command")
+        except APIError as e:
+            return await self.error_handler(ctx, e)
```

Comment:
since those two methods are unused you can remove them (except the one in guild default command, but just put it in it's body)

Reply:
Removed the uneeded funtions and included guildid_to_guildname in !guild default's body